std/range/package.d

```diff
@@ -7125,9 +7131,57 @@ private:
     size_t _chunkSize;
 }
 
+/// ditto
+struct Chunks(Source)
+if (isInputRange!Source && !isForwardRange!Source)
+{
+    static struct Chunk
+    {
+        private Chunks* impl;
+
+        @property bool empty() { return impl.curSizeLeft == 0 || impl.r.empty; }
+        @property auto front() { return impl.r.front; }
+        void popFront()
+        {
+            assert(impl.curSizeLeft > 0 && !impl.r.empty);
+            impl.curSizeLeft--;
+            impl.r.popFront();
+        }
+    }
+
+    private Source r;
+    private size_t chunkSize;
+    private size_t curSizeLeft;
+    bool empty;
+
+    this(Source _r, size_t _chunkSize)
+    {
+        r = _r;
+        chunkSize = _chunkSize;
+        curSizeLeft = chunkSize;
+        empty = r.empty;
+    }
+
+    @property Chunk front() { return Chunk(&this); }
```

Comment:
OK, just reworked the code a bit to move the implementation bits into a heap-allocated struct, so that the outer `Chunks` is just a wrapper around a pointer to it. This makes it cleaner and also fixes the problems with `-dip25 -dip1000`.